pymodbus/transaction.py

```diff
@@ -463,13 +468,17 @@ def _recv(self, expected_response_length, full) -> bytes:  # noqa: C901
                         expected_response_length -= min_size
                         total = expected_response_length + min_size
                 else:
+                    if exception_length is None:
```

Comment:
Not covered by test.

Reply:
This was flagged by the type-checker, if `exception_length` is `None` we would try to subtract from `exception_length` in `expected_response_length = exception_length - min_size` which would not work.

Reply:
What has that to do with that you have not added test cases, that show why the old code was broken and the new code works.

Reply:
> What has that to do with that you have not added test cases, that show why the old code was broken and the new code works.

Tweaked this a bit more to get it passing the type checker a slightly different way, I'm not sure how we would actually hit this codepath however so a bit hard to write tests.